enumerables.rb

```diff
@@ -0,0 +1,190 @@
+    return to_enum(:my_each_with_index) unless block_given?
+
+    counter = 0
+    size = length - 1
+
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    required = []
+
+    my_each do |r|
+      required.push(r) if yield(r)
+    end
+    required
+  end
+
+  def my_all?(*param)
+    track_negative = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size && !self[counter + 1].nil?
+      if param[0].class == Class
+        track_negative += self[counter].class == self[counter + 1].class ? 0 : 1
+        counter += 1
+      elsif param[0].class == Regexp
+        track_negative += self[counter].to_s.match(param[0]).nil? == self[counter + 1].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |p|
+          track_negative += yield(p) ? 0 : 1
+          counter += 1
+        end
+      elsif param[0].nil? && !block_given?
+        my_each do |ds|
+          track_negative += ds ? 0 : 1
+          counter += 1
+        end
+      end
+    end
+    !track_negative.positive?
+  end
+
+  def my_any?(*param)
+    yes = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size
+      if param[0].class == Class
+        yes += self[counter].class == param[0] ? 1 : 0
+        counter += 1
+      elsif param[0].class == Regexp
+        yes += self[counter].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |sth|
+          yes += yield(sth) ? 1 : 0
+          counter += 1
+        end
+      else
+        my_each do |sth|
+          yes += sth ? 1 : 0
+          counter += 1
+        end
+      end
+      yes
+    end
+    yes.positive?
+  end
+
+  def my_none?(*param)
+    satisfies = 0
+    size = count - 1
+    counter = 0
+
+    while counter <= size
+      if param[0].class == Class
+        satisfies += self[counter].class == param[0] ? 1 : 0
+        counter += 1
+      elsif param[0].class == Regexp
+        satisfies += self[counter].to_s.match(param[0]).nil? ? 0 : 1
+        counter += 1
+      elsif block_given?
+        my_each do |s|
+          satisfies += yield(s) ? 1 : 0
+          counter += 1
+        end
+      else
+        my_each do |t|
+          satisfies += t ? 1 : 0
+          counter += 1
+        end
+      end
+    end
+    !satisfies.positive?
+  end
+
+  def my_count(*param)
+    counter = 0
+    return to_a.length if param.empty? && !block_given?
+
+    if param && !block_given?
+      my_each do |d|
+        param[0] == d ? counter += 1 : false
+      end
+    elsif block_given?
+      my_each do |el|
+        yield(el) ? counter += 1 : false
+      end
+    end
+    counter
+  end
+
+  def my_map
+    return self unless block_given?
+
+    cont_arr = []
+
+    my_each do |cont|
+      cont_arr.push(yield(cont)) if yield(cont)
+    end
+    cont_arr
+  end
+
```

Comment:
- my_map returns an Enumerator if no block is given:
`p [1, 2, 3].my_map should return <Enumerator>`
-  my_map returns a new array with the results of running block once for every element in enum.
##### For example
`block =  proc { |num| num < (4 + 3) / 2 }`
`p [1,22,34,5].my_map(&block) ==  [true]`
`p [1,22,34,5].map(&block) == [true, false, false, false]`